server/keyspace/keyspace_test.go

```diff
@@ -0,0 +1,229 @@
+	for i, createRequest := range requests {
+		_, err := manager.CreateKeyspace(createRequest)
+		re.NoError(err)
+		updateRequest := &UpdateKeyspaceRequest{
+			Name:        createRequest.Name,
+			UpdateState: true,
+			NewState:    keyspacepb.KeyspaceState(rand.Int31n(3)),
+			Now:         time.Now(),
+			ToPut: map[string]string{
+				"config_entry_1":    strconv.Itoa(i),
+				"config_entry_2":    strconv.Itoa(i),
+				"additional_config": strconv.Itoa(i),
+			},
+			ToDelete: []string{"config_entry_2"},
+		}
+
+		updated, err := manager.UpdateKeyspace(updateRequest)
+		re.NoError(err)
+		matchUpdateRequest(re, updateRequest, updated)
+	}
+}
+
+func TestLoadRangeKeyspace(t *testing.T) {
+	re := require.New(t)
+	manager := newKeyspaceManager()
+	requests := createKeyspaceRequests(10)
+
+	startID := 100
+	// force keyspace id start with startID
+	for i := 0; i < startID-1; i++ {
+		_, _ = manager.idAllocator.Alloc()
+	}
+	for _, createRequest := range requests {
+		_, err := manager.CreateKeyspace(createRequest)
+		re.NoError(err)
+	}
+
+	// load all of them
+	keyspaces, err := manager.LoadRangeKeyspace(0, 0)
+	re.NoError(err)
+	re.Equal(10, len(keyspaces))
+	for i := range keyspaces {
+		re.Equal(uint32(startID+i), keyspaces[i].Id)
+		matchCreateRequest(re, requests[i], keyspaces[i])
+	}
+
+	// load first 3
+	keyspaces, err = manager.LoadRangeKeyspace(0, 3)
+	re.NoError(err)
+	re.Equal(3, len(keyspaces))
+	for i := range keyspaces {
+		re.Equal(uint32(startID+i), keyspaces[i].Id)
+		matchCreateRequest(re, requests[i], keyspaces[i])
+	}
+
+	// load 3 starting with startID + 5
+	loadStart := startID + 5
+	keyspaces, err = manager.LoadRangeKeyspace(uint32(loadStart), 3)
+	re.NoError(err)
+	re.Equal(3, len(keyspaces))
+	for i := range keyspaces {
+		re.Equal(uint32(loadStart+i), keyspaces[i].Id)
+		matchCreateRequest(re, requests[i+5], keyspaces[i])
+	}
+}
+
+// TestUpdateMultipleKeyspace checks that updating multiple keyspace's config simultaneously
+// will be successful.
+func TestUpdateMultipleKeyspace(t *testing.T) {
+	re := require.New(t)
+	manager := newKeyspaceManager()
+	requests := createKeyspaceRequests(5)
+	for _, createRequest := range requests {
+		_, err := manager.CreateKeyspace(createRequest)
+		re.NoError(err)
+	}
+
+	// concurrently update all keyspaces' testConfig sequentially.
+	end := 100
+	wg := sync.WaitGroup{}
+	for _, request := range requests {
+		wg.Add(1)
+		go func(name string) {
+			defer wg.Done()
+			updateKeyspaceConfig(re, manager, name, end)
+		}(request.Name)
+	}
+	wg.Wait()
+
+	// check that eventually all test keyspaces' test config reaches end
+	for _, request := range requests {
+		keyspace, err := manager.LoadKeyspace(request.Name)
+		re.NoError(err)
+		re.Equal(keyspace.Config[testConfig], strconv.Itoa(end))
+	}
+}
+
+// matchCreateRequest verifies a keyspace meta matches a create request.
+func matchCreateRequest(re *require.Assertions, request *CreateKeyspaceRequest, meta *keyspacepb.KeyspaceMeta) {
```

Comment:
It can be named ```checkKeyspace```. Like https://github.com/tikv/pd/blob/34a4cce2cb07f8aa4283e28ba40a667e1321a1fb/server/cluster/cluster_test.go#L1763